posts/2021-03-24-whats-new-in-MP-Rest-Client2.0.adoc

```diff
@@ -0,0 +1,212 @@
+</dependency>
+----
+
+or, if you use link:{url-prefix}/guides/gradle-intro.html[Gradle]:
+[source,gradle]
+----
+dependencies {
+    mpRestClient group: 'org.eclipse.microprofile.rest.client', name: 'microprofile-rest-client-api', version: '2.0'
+}
+----
+
+Also, make sure to configure your Liberty server with the `mpRestClient-2.0` feature in the server.xml:
+[source,xml]
+----
+<server>
+  <featureManager>
+    <feature>mpRestClient-2.0</feature>
+    <!-- ... -->
+  </featureManager>
+  <!-- ... -->
+</server>
+----
+
+That's it! Now that we've got our development and deployment environments set up, it's time to code!
+
+
+[#query-param-style]
+== Using QueryParamStyle to specify how collections of query parameters should be formatted
+
+Rest Client interfaces can specify query parameters using the `@QueryParam("paramName")` annotation, but often a server
+will require that multi-valued query parameters must be formatted in a certain way. For example, suppose we have a
+client interface like:
+[source,java]
+----
+@RegisterRestClient
+public interface MyClient {
+    @GET
+    String multiValues(@QueryParam("myParam") List<String> values);
+}
+----
+By default, if a caller invokes the the client with `multiValues(Arrays.asList("a", "b", "c"))` the MP Rest Client will
+produce an HTTP request with multiple key/value pairs - something like: `?myParam=a&myParam=b&myParam=c`
+
+Most servers will handle that just fine, but other servers will require that the HTTP request be a single key with a 
+comma-separated list of values, like: `?myParam=a,b,c`
+
+Other servers will require array-like syntax, such as: `?myParam[]=a&myParam[]=b&myParam[]=c`
+
+In order to support those other server types, you can now use the `QueryParamStyle` enum when building the client
+instance - for example:
+
+[source,java]
+----
+MyClient client = RestClientBuilder.newBuilder()
+                                   .queryParamStyle(QueryParamStyle.COMMA_SEPARATED)
+                                   //...
+                                   .build(MyClient.class);
+----
+
+Alternatively the query parameter style can be declared through MP Config using a property like:
+
+`com.mypkg.MyClient/mp-rest/queryParamStyle=ARRAY_PAIRS`
+
+.QueryParamStyle Options
+|=======================
+|Enum Value            | Output Example
+|MULTI_PAIRS (default) | `?myParam=a&myParam=b&myParam=c`
+|COMMA_SEPARATED       | `?myParam=a,b,c`
+|ARRAY_PAIRS           | `?myParam[]=a&myParam[]=b&myParam[]=c`
+|=======================
+
+[#proxy-server]
+== Proxy server support
+
+Many RESTful endpoints might require the use of a proxy server to access. MP Rest Client 2.0 makes it easier
+and more portable to specify a proxy server with the new `proxyAddress(host, port)` method on the `RestClientBuilder`
+class. For example, suppose you need to access an endpoint via a proxy server at myproxy.xyz.com on port 1080. You could
+build your rest client instance with code like:
```

Comment:
should these values be monospaced?

suppose you need to access an endpoint via a proxy server at `myproxy.xyz.com` on port `1080`.